gui/wxpython/startup/guiutils.py

```diff
@@ -215,6 +215,7 @@ def rename_mapset_interactively(guiparent, grassdb, location, mapset):
     """
     Rename selected mapset
     """
+    newmapset = None
```

Comment:
The result of `dlg.GetValue()` still gets returned even when `rename_mapset()` fails. You need to do `newmapset = None` in the `except` clause to signal that the mapset was renamed, i.e., there is no new name to report (assuming that the caught exception here means no renaming happened).

The same issue(s) as here in `rename_mapset_interactively()` seems to be present in `rename_location_interactively()`.

Reply:
I added newmapset = None into except clause in rename_mapset_interactively and newlocation = None into except clause in rename_location_interactively.